node/node_newblock.go

```diff
@@ -89,23 +93,30 @@ func (node *Node) proposeNewBlock() (*types.Block, error) {
 		err         error
 	)
 
-	node.Worker.GetCurrentHeader().SetCoinbase(coinbase)
-
+	header := node.Worker.GetCurrentHeader()
 	// After staking, all coinbase will be the address of bls pub key
-	if header := node.Worker.GetCurrentHeader(); node.Blockchain().Config().IsStaking(header.Epoch()) {
-		addr := common.Address{}
+	if node.Blockchain().Config().IsStaking(header.Epoch()) {
 		blsPubKeyBytes := node.Consensus.LeaderPubKey.GetAddress()
```

Comment:
This will fail in case of multi-bls scenario, when the node has multiple bls keys and corresponding addresses. Should we fetch the address from node.Consensus.SelfAddresses map?

Reply:
nop